### PR TITLE
Close #22305: Add tab state into the TabsTrayStore 

### LINKED ISSUE #22305: Move tabs logic state to TabsTrayStore
We do a lot of work in the `TabSorter` that talks directly to the UI pieces. Every other interactor/controller or logical piece needs to re-calculate all of those sections if they want to use them too and we end up duplicating lots of settings checks and logic in those places too.

We can do better by letting the TabSorter apply the appropriate logic, and putting the final list of tabs into the `TabsTrayStore` for everyone to observe and use where needed.

### PR DESCRIPTION
**Pull request by jonalmeida**
TL;DR: the simplest way to understand what this change does is to read `TabSorterTest`.

## Problem
1. In our `TabSorter`, we are doing many things:
   1. Organizing tabs in different sections.
   2. Updating the UI adapters for those sections.
   3. Telemetry.
2. For other parts of the code base, we re-calculate the tabs for each section in those individual places. This is a problem because there is duplication, room for error, and we have to pass things like `context.settings()` everywhere to know what the right state should be.
3. We're unable to write simple tests because of the interchanging UI components.

In this change, we put the tabs into the `TabsTrayStore` so that we have one place to hold all the final state. This means we only have to access settings once when we do the sorting and all other observers can react accordingly.

## AC partitions

When AC adds support for our various groupings, we can then remove `TabSorter` and use the APIs to feed the tabs into `TabsTrayStore` without needing to change the implementation of any UI components.

## What does this mean for Compose?

With the tabs in the store, we have existing helper extensions for compose made specifically for working on `Store` so we can leverage those and have a straight-forward list to consume in Compose. For example:
```kotlin
fun BrowserList(store: TabsTrayStore) {
  val inactiveTabs = store.observeAsComposableState { state -> state.inactiveTabs }
  items(inactiveTabs) { tab ->
    InactiveTab(tab)
  }
  // etc..
}
```

See https://github.com/mozilla-mobile/fenix/issues/21898, https://github.com/mozilla-mobile/fenix/issues/21899, https://github.com/mozilla-mobile/fenix/issues/21901 for when this will help.

---

Some simple next things to do:
 - Remove the telemetry from `TabSorter` as well and put it in a `Middleware` on the `TabsTrayStore`.
 - Move some of the remaining bits of code that calculate inactive tabs manually to use the store.
 - Move the `InactiveTabsInteractor` out of the UI class too.
 - Write more test for `TabSorter` where needed.
 - Refactor the `scrollToTab` to use the store to know how to scroll (then write tests for that too!)

I've also sneaked in one commit that corrects the naming convention of a `TabsTrayAction`.

cc: @csadilek @MozillaNoah 

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by rocketsroger — APPROVED** [review]
🚢

**Review by Amejia481 — APPROVED** [review]
🚢 !

**Comment by jonalmeida** [comment]
Thank you for reviews!
